Source/WebCore/Modules/mediastream/ImageCapture.cpp

```diff
@@ -81,35 +81,42 @@ void ImageCapture::takePhoto(PhotoSettings&& settings, DOMPromiseDeferred<IDLInt
     });
 }
 
-void ImageCapture::getPhotoCapabilities(PhotoCapabilitiesPromise&& promise)
+void ImageCapture::getPhotoCapabilities(DOMPromiseDeferred<IDLDictionary<PhotoCapabilities>>&& promise)
 {
     auto identifier = LOGIDENTIFIER;
     ALWAYS_LOG(identifier);
 
-    if (m_track->readyState() == MediaStreamTrack::State::Ended) {
-        ERROR_LOG(identifier, "rejecting promise, track has ended");
-        promise.reject(Exception { ExceptionCode::InvalidStateError, "Track has ended"_s });
-        return;
-    }
+    m_track->getPhotoCapabilities()->whenSettled(RunLoop::main(), [this, protectedThis = Ref { *this }, promise = WTFMove(promise), identifier = WTFMove(identifier)] (auto&& result) mutable {
+        queueTaskKeepingObjectAlive(*this, TaskSource::ImageCapture, [this, promise = WTFMove(promise), result = WTFMove(result), protectedThis, identifier = WTFMove(identifier)] () mutable {
```

Comment:
Capturing `protectedThis` is redundant here. The function you're calling is named queueTask*KeepingObjectAlive*

Reply:
Indeed 😳